tests/rptest/tests/nodes_decommissioning_test.py

```diff
@@ -101,3 +109,121 @@ def node_removed():
         wait_until(node_removed, timeout_sec=120, backoff_sec=2)
 
         self.run_validation(enable_idempotence=False, consumer_timeout_sec=45)
+
+
+class SINodesDecommissioningTest(EndToEndTest):
+    """
+    Enable SI on a topic and test node decommissioning
+    """
+    segment_size = 1048576  # 1MB
+    s3_host_name = "minio-s3"
+    s3_access_key = "panda-user"
+    s3_secret_key = "panda-secret"
+    s3_region = "panda-region"
+    s3_topic_name = "panda-topic"
+    topics = []
+
+    def __init__(self, test_context):
+        super(SINodesDecommissioningTest,
+              self).__init__(test_context=test_context)
+
+        self.s3_bucket_name = f"panda-bucket-{uuid.uuid1()}"
+        self._extra_rp_conf = dict(
+            cloud_storage_enabled=True,
+            cloud_storage_enable_remote_read=True,
+            cloud_storage_enable_remote_write=True,
+            cloud_storage_access_key=self.s3_access_key,
+            cloud_storage_secret_key=self.s3_secret_key,
+            cloud_storage_region=self.s3_region,
+            cloud_storage_bucket=self.s3_bucket_name,
+            cloud_storage_disable_tls=True,
+            cloud_storage_api_endpoint=self.s3_host_name,
+            cloud_storage_api_endpoint_port=9000,
+            cloud_storage_reconciliation_interval_ms=500,
+            cloud_storage_max_connections=5,
+            log_segment_size=self.segment_size,  # 1MB
+            cloud_storage_cache_size=5 * self.segment_size,
+        )
+
+        self.scale = Scale(test_context)
+        self.redpanda = RedpandaService(
+            context=test_context,
+            num_brokers=4,
+            extra_rp_conf=self._extra_rp_conf,
+        )
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.s3_client = S3Client(
+            region=self.s3_region,
+            access_key=self.s3_access_key,
+            secret_key=self.s3_secret_key,
+            endpoint=f"http://{self.s3_host_name}:9000",
+            logger=self.logger,
+        )
+
+    def setUp(self):
+        self.s3_client.empty_bucket(self.s3_bucket_name)
+        self.s3_client.create_bucket(self.s3_bucket_name)
+        self.redpanda.start()
+
+        for partition_count in range(1, 5):
+            for replication_factor in (3, 3):
```

Comment:
The `test_decommissioning_crashed_node_*` tests were failing with (1,3) but succeed with (3,3).
Why would this be? CC: @mmaslankaprv 

I moved this change to a separate PR so it'll be easier to track in a GH issue. See #4115

Reply:
I think we need to simply remove `for` loop from there. The problem is that with single replica topics we can not decommission random node that crashed as it may host that single replica and decommissioning process will never finish.